web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/TraceContextCorrelation.java

```diff
@@ -0,0 +1,288 @@
+package com.microsoft.applicationinsights.web.internal.correlation;
+
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.RequestTelemetry;
+import com.microsoft.applicationinsights.web.internal.ThreadContext;
+import com.microsoft.applicationinsights.web.internal.correlation.tracecontext.Traceparent;
+import com.microsoft.applicationinsights.web.internal.correlation.tracecontext.Tracestate;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * A class that is responsible for performing correlation based on W3C protocol.
+ * This is a clean implementation of W3C protocol and doesn't have the backward
+ * compatibility with AI-RequestId protocol.
+ *
+ * @author Dhaval Doshi
+ */
+public class TraceContextCorrelation {
+
+    public static final String CORRELATION_HEADER_NAME = "traceparent";
+    public static final String CORRELATION_COMPANION_HEADER_NAME = "tracestate";
+    public static final String AZURE_TRACEPARENT_COMPONENT_INITIAL = "az";
+
+    /**
+     * Private constructor as we don't expect to create an object of this class.
+     */
+    private TraceContextCorrelation() {}
+
+    /**
+     * This method is responsible to perform correlation for incoming request by populating it's
+     * traceId, spanId and parentId. It also stores incoming tracestate into ThreadLocal for downstream
+     * propagation.
+     * @param request
+     * @param response
+     * @param requestTelemetry
+     */
+    public static void resolveCorrelation(HttpServletRequest request, HttpServletResponse response,
+        RequestTelemetry requestTelemetry) {
+
+        try {
+            if (request == null) {
+                InternalLogger.INSTANCE.error("Failed to resolve correlation. request is null.");
+                return;
+            }
+
+            if (response == null) {
+                InternalLogger.INSTANCE.error("Failed to resolve correlation. response is null.");
+                return;
+            }
+
+            if (requestTelemetry == null) {
+                InternalLogger.INSTANCE.error("Failed to resolve correlation. requestTelemetry is null.");
+                return;
+            }
+
+
+            String traceId = request.getHeader(CORRELATION_HEADER_NAME);
+
+            Traceparent outGoingTraceParent = null;
+
+            if (traceId == null || traceId.isEmpty()) {
+                outGoingTraceParent = new Traceparent();
+
+                // represents the id of the current request.
+                requestTelemetry.setId(outGoingTraceParent.getTraceId() + "-" + outGoingTraceParent.getSpanId());
```

Comment:
|outGoingTraceParent.getTraceId().outGoingTraceParent.getSpanId().

Reply:
I know these are backward compatibility comments. Understand it's importance. The question I have is, should we make changes in UX to understand correct format too or should SDK set it as UX expects? 
My understanding is that the protocol is what defines the contract between UX and SDK.